compiling current git master snapshot fails on openSUSE

The DBLD CentOS 7 build worked fine, but when I tried to build on openSUSE 15.2 from the same source tgz, it failed:

```
[   73s] libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I./lib -I./modules -I./lib -I./modules -I./modules/syslogformat -I./modules/syslogformat -pthread -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_28 -DGLIB_VERSION_MAX_ALLOWED=0x023C00 -I./lib/eventlog/src -I./lib/eventlog/src -DOPENSSL_LOAD_CONF -D_BSD_SOURCE -D__BSD_SOURCE -D__FAVOR_BSD -DHAVE_NET_ETHERNET_H -D_DEFAULT_SOURCE -I/usr/include/dbi -I/usr/include/dbi/dbi -isystem ./lib/ivykis/src/include -isystem ./lib/ivykis/src/include -D_GNU_SOURCE -D_DEFAULT_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wshadow -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros -O2 -g -pthread -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -c modules/syslogformat/syslog-format.c  -fPIC -DPIC -o modules/syslogformat/.libs/libsyslogformat_la-syslog-format.o
[   73s] In file included from lib/logmsg/tests/dump_logmsg.c:23:0:
[   73s] lib/logmsg/tests/test_logmsg_serialize.c:32:10: fatal error: criterion/criterion.h: No such file or directory
[   73s]  #include <criterion/criterion.h>
[   73s]           ^~~~~~~~~~~~~~~~~~~~~~~
[   73s] compilation terminated.
[   73s] make[2]: *** [Makefile:20746: lib/logmsg/tests/dump_logmsg-dump_logmsg.o] Error 1
[   73s] make[2]: *** Waiting for unfinished jobs....
[   74s] make[1]: *** [Makefile:24490: all-recursive] Error 1
[   74s] make: *** [Makefile:10564: all] Error 2
[   74s] error: Bad exit status from /var/tmp/rpm-tmp.OcGbcz (%build)
```

I suspect, that it might be related to https://github.com/syslog-ng/syslog-ng/pull/3799 by @bazsi

Tested the same source tgz on Fedora 34, without using DBLD, and ran into the same problem:
```
  CC       lib/logmsg/tests/dump_logmsg-dump_logmsg.o
  CC       modules/syslogformat/libsyslogformat_la-syslog-format.lo
In file included from lib/logmsg/tests/dump_logmsg.c:23:
lib/logmsg/tests/test_logmsg_serialize.c:32:10: fatal error: criterion/criterion.h: No such file or directory
   32 | #include <criterion/criterion.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:20746: lib/logmsg/tests/dump_logmsg-dump_logmsg.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:24490: all-recursive] Error 1
make: *** [Makefile:10564: all] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.zk10bB (%build)
    Bad exit status from /var/tmp/rpm-tmp.zk10bB (%build)
```

--- a/lib/logmsg/tests/Makefile.am
+++ b/lib/logmsg/tests/Makefile.am
@@ -38,6 +38,7 @@ lib_logmsg_tests_test_logmsg_ack_CFLAGS = $(TEST_CFLAGS)
 lib_logmsg_tests_test_nvhandle_desc_array_LDADD = $(TEST_LDADD)
 lib_logmsg_tests_test_nvhandle_desc_array_CFLAGS = $(TEST_CFLAGS)
 
+if ENABLE_TESTING
 noinst_PROGRAMS +=				\
 	lib/logmsg/tests/dump_logmsg
 
@@ -49,3 +50,5 @@ dump-logmsg: lib/logmsg/tests/dump_logmsg
 	DIR=lib/logmsg/tests/messages/ && \
 	mkdir -p $${DIR} && \
 	lib/logmsg/tests/dump_logmsg $(VERSION) > $${DIR}/syslog-ng-$(VERSION)-msg.h
+
+endif

